addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -2277,6 +2277,123 @@ QUnit.module('relational_fields', {
         });
     });
 
+    QUnit.test('pressing TAB on a \'no_quick_create\' many2one should not trigger M2ODialog', function (assert) {
+        // most features of the ENTER KEY should be tested for the TAB key
+        // its behaviour is more blurry, since it is partly performative and partly navigation
+        var done = assert.async();
+        assert.expect(2);
+
+        var M2O_DELAY = relationalFields.FieldMany2One.prototype.AUTOCOMPLETE_DELAY;
+        relationalFields.FieldMany2One.prototype.AUTOCOMPLETE_DELAY = 0;
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form>' +
+                    '<field name="trululu" options="{\'no_quick_create\': True}"/>' +
+                    '<field name="foo"/>' +
+                '</form>',
+            archs: {
+                'partner,false,form': '<form string="Partners"><field name="display_name"/></form>',
+            },
+        });
+
+        var $input = form.$('.o_field_many2one input');
+        $input.val("Something that does not exist").trigger('input');
+        $('.ui-autocomplete .ui-menu-item a:contains(Create and)').trigger('mouseenter');
+        concurrency.delay(0).then(function() {
+            $input.trigger($.Event('keydown', {
+                which: $.ui.keyCode.TAB,
+                keyCode: $.ui.keyCode.TAB,
+            }));
+            $input.trigger($.Event('keypress', {
+                which: $.ui.keyCode.TAB,
+                keyCode: $.ui.keyCode.TAB,
+            }));
+            $input.trigger($.Event('keyup', {
+                which: $.ui.keyCode.TAB,
+                keyCode: $.ui.keyCode.TAB,
+            }));
+            concurrency.delay(0).then(function() {
+                $input.blur();
+                assert.strictEqual($('.modal').length, 1,
+                    "should have one modal in body");
```

Comment:
i'd add an assert here, checking that "Something that does not exist" is correctly set by default in the modal